From 65b0e4973a2957d79b3132fdb94c303cd59b2836 Mon Sep 17 00:00:00 2001
From: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Date: Thu, 7 Mar 2019 13:11:25 +0700
Subject: [PATCH 751/909] iommu/ipmmu-vmsa: Bypass whitelist check for
 unsupported SoCs

commit adc2d8f520e43ae9345d68363087bbd1149f4623 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Whitelist check is specific for Renesas R-Car Gen3 SoCs only.
Bypass the check for other SoCs.

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/iommu/ipmmu-vmsa.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/iommu/ipmmu-vmsa.c b/drivers/iommu/ipmmu-vmsa.c
index f9a6bf186c93..e0b832007a1f 100644
--- a/drivers/iommu/ipmmu-vmsa.c
+++ b/drivers/iommu/ipmmu-vmsa.c
@@ -1080,6 +1080,13 @@ static bool ipmmu_slave_whitelist(struct device *dev, u32 *ids)
 		goto exit;
 	}
 
+	/*
+	 * For R-Car Gen3 SoCs, use a white list to check the uTLB set up
+	 * For other SoCs, return true
+	 */
+	if (!mmu->features->whitelist)
+		return true;
+
 	if (!mmu->whitelist[0]) {
 		pr_debug("%s Whitelist not found on %s!!!\n",
 			 __func__, dev_name(mmu->dev));
-- 
2.17.1

